secrets/src/int/classify_public.rs

```diff
@@ -0,0 +1,44 @@
+/// This file defines functions for classifying and declassifying various types.
+/// We give definitions for all conversions so that they can be tested.
+/// We define no-ops here and force inlining, to ensure that these are free.
+use crate::traits::*;
+
+// TODO: remove hax exemptions once this is supported
+
+// Immutable references to slices can be classified
+#[cfg(not(hax))]
+impl<'a, T: Scalar> ClassifyRef for &'a [T] {
+    type ClassifiedRef = &'a [T];
+    #[inline(always)]
+    fn classify_ref(self) -> &'a [T] {
+        self
+    }
+}
+
+// Immutable references to slices can be declassified
+impl<'a, T: Scalar> DeclassifyRef for &'a [T] {
+    type DeclassifiedRef = &'a [T];
+    fn declassify_ref(self) -> &'a [T] {
```

Comment:
Should this one get inlined as well? (I'd expect the compiler to do that anyway though.)